src/custom/state/lists/hooks/hooksMod.ts

```diff
@@ -157,9 +158,11 @@ export function useActiveListUrls(): string[] | undefined {
   // MOD: adds { chainId } support to the hooks
   const { chainId: connectedChainId } = useWeb3React()
   const chainId = supportedChainId(connectedChainId) ?? DEFAULT_NETWORK_FOR_LISTS
-  return useAppSelector((state) => state.lists[chainId]?.activeListUrls)?.filter(
-    (url) => !UNSUPPORTED_LIST_URLS[chainId]?.includes(url)
-  )
+  const activeListUrls = useAppSelector((state) => state.lists[chainId]?.activeListUrls, shallowEqual)
+
+  return useMemo(() => {
+    return activeListUrls?.filter((url) => !UNSUPPORTED_LIST_URLS[chainId]?.includes(url))
+  }, [chainId, activeListUrls])
```

Comment:
Could this by any chance fix the issues we have with the coingecko list?

For reference https://github.com/cowprotocol/cowswap/pull/719

@ramirotw @henrypalacios keep this in mind for the coingecko task

Reply:
Unfortunately, I don't know about the coingecko list issue :(
This is just performance optimization. Ideally, we must refactor the main part of tokens-list-related code.

Reply:
No worries, they are aware.
Just reminding them in case they are touching the same area, to be aware of your changes